pyvo/mivot/writer/instances_from_models.py

```diff
@@ -0,0 +1,722 @@
+
+    def extract_epoch_position_parameters(self):
+        """
+        Build a dictionary with the 3 parameters required by
+        :py:meth:`pyvo.mivot.writer.InstancesFromModels.add_mango_epoch_position`
+        (frames, mapping, semantic).
+
+        - frames: references to the frames created from COOSYS/TIMSYS
+          (see :py:meth:`pyvo.mivot.writer.InstancesFromModels.extract_frames`
+        - mapping: inferred from the FIELD ucd-s
+        - semantics: hard-coded
+
+        This method does not add the ``EpochPosition`` property because the
+        mapping parameters often need first to be completed (or fixed) by hand.
+        This especially true for the correlations that cannot be automatically
+        detected in a generic way (no specific UCD)
+
+        .. code-block::
+
+            epoch_position_mapping = builder.extract_epoch_position_parameters()
+            builder.add_mango_epoch_position(**epoch_position_mapping)
+
+        Returns
+        -------
+        {"frames", "mapping", "semantics"}
+            Dictionary of mapping elements that can unpacked to feed ``add_mango_epoch_position()``.
+        """
+        mapping, error_mapping = self._header_mapper.extract_epochposition_mapping()
+        mapping["errors"] = error_mapping
+        mapping["correlations"] = {}
+        semantics = {"description": "6 parameters position",
+                     "uri": "https://www.ivoa.net/rdf/uat/2024-06-25/uat.html#astronomical-location",
+                     "label": "Astronomical location"}
+        frames = self.extract_frames()
+        return {"frames": frames, "mapping": mapping, "semantics": semantics}
+
+    def add_photcal(self, filter_name):
+        """
+        Add to the GLOBALS the requested photometric calibration as defined in PhotDM1.1
+
+        The MIVOT serialization is provided by the SVO Filter Profile Service
+        (https://ui.adsabs.harvard.edu/abs/2020sea..confE.182R/abstract)
+
+        It is returned as one block containing the whole PhotCal instance.
+        The filter instance is extracted from the PhotCal object, where it is
+        replaced with a REFERENCE.
+        This makes the filter available as a GLOBALS to objects that need it.
+
+        - The dmid of the PhotCal is ``_photcal_DMID`` where DMID is the formatted
+          version of ``filter_name``
+          (see `pyvo.mivot.utils.MivotUtils.format_dmid`).
+        - The dmid of the PhotFilter is ``_photfilter_DMID`` where DMID is the formatted
+          version of ``filter_name``.
+
+        Parameters
+        ----------
+        filter_name : str
+            FPS identifier (SVO Photcal ID) of the request filter
+
+        Returns
+        -------
+        (str, str)
+            dmids of photcal and filter instances
+
+        """
+        response = requests.get(Url.FPS + filter_name)
+        if (http_code := response.status_code) != 200:
+            logging.error("FPS service error: %s", http_code)
```

Comment:
nitpicking, but all of these (there are a couple of other similar ones) could be f-strings.

Reply:
Oh, indeed, you're absolutely right. Please keep all as is.